### PR TITLE
refactor(homepage): refactor data models for homepage

### PR DESCRIPTION
**Pull request by tensan8**
API integration using typescript

### CODE REVIEW

**Comment by tensan8** [comment]
Issues for ac3fbee: data can be retrieved from API but cannot be accessed using the Result VO object. ViewModel has not been implemented.

image

File: src/app/Homepage/mappers/cardMapper.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { PostModel } from '../../../domain/models/Post'
+import * as React from 'react'
+import Card from '../../common/atoms/card'
+import SummaryCardContent from '../../common/atoms/SummaryCardContent'
+import { UserModel } from '../../../domain/models/User'
+import { CommentModel } from '../../../domain/models/Comment'
+
+export const MapPostToCardList = (
```
Comment (igarramaddhan) [inline src/app/Homepage/mappers/cardMapper.tsx:8]:
```suggestion
export const PostCardList = (
```

Comment (igarramaddhan) [reply src/app/Homepage/mappers/cardMapper.tsx:8]: don't forget to change the implementation

File: src/app/Homepage/pages/index.tsx [hunk 2]
```diff
@@ -11,25 +13,20 @@ const Homepage = (): JSX.Element => {
...
-                        content='Content'
-                        username='Username'
-                        company='Company'
-                        numberOfComment={0}
-                    />
-                }
-                clickable={true}
-                onCardClick={handleHomepageCardClicked}
-            />
+            {allPosts != null && allUsers != null && allComments != null && allUsers.length !== 0
```
Comment (igarramaddhan) [inline src/app/Homepage/pages/index.tsx:26]: move this implementation to `PostCardList`

Comment (igarramaddhan) [reply src/app/Homepage/pages/index.tsx:26]:
```suggestion
            <PostCardList
                allPosts={allPosts}
                handleHomepageCardClicked={handleHomepageCardClicked}
                allUsers={allUsers}
                allComents={allComments}
             />
```

File: src/app/Homepage/mappers/cardMapper.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  postList: PostModel[],
+  clickCallback: () => any,
+  usersList: UserModel[],
+  commentsList: CommentModel[]): any => {
+  const userLookup = (postUserId: number): any => {
```
Comment (igarramaddhan) [inline src/app/Homepage/mappers/cardMapper.tsx:13]: implement `useCallback`

File: src/app/Homepage/mappers/cardMapper.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    for (let index = 0; index < usersList.length; index++) {
+      if (usersList[index].id === postUserId) {
+        return usersList[index]
+      }
+    }
+  }
+
+  const commentCounter = (postId: number): number => {
```
Comment (igarramaddhan) [inline src/app/Homepage/mappers/cardMapper.tsx:21]: implement `useCallback`

File: src/app/common/molecules/profileHead.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+// import UserIcon from '../assets/user.png'
```
Comment (igarramaddhan) [inline src/app/common/molecules/profileHead.jsx:1]: remove if not used

Comment (tensan8) [reply src/app/common/molecules/profileHead.jsx:1]: If I am not mistaken this one will be refactored and used later on the other page, if it is not an issue to keep it there for now then I think let us just keep it there

File: src/app/Homepage/mappers/cardMapper.tsx (hunk 4 above)
Comment (tensan8) [reply src/app/Homepage/mappers/cardMapper.tsx]: Implemented on 8cd5fbe

File: src/app/Homepage/mappers/cardMapper.tsx (hunk 3 above)
Comment (tensan8) [reply src/app/Homepage/mappers/cardMapper.tsx]: Implemented on 8cd5fbe

File: src/app/Homepage/mappers/cardMapper.tsx (hunk 1 above)
Comment (tensan8) [reply src/app/Homepage/mappers/cardMapper.tsx]: Implemented on 8cd5fbe

File: src/app/Homepage/pages/index.tsx (hunk 2 above)
Comment (tensan8) [reply src/app/Homepage/pages/index.tsx]: Implemented on 8cd5fbe

**Review by igarramaddhan — APPROVED** [review]

**Pull request merged**